src/PluginsSystem/Microsoft.Performance.Toolkit.Plugins.Core/Discovery/IPluginDiscoverer.cs

```diff
@@ -0,0 +1,68 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Performance.SDK.Processing;
+using Microsoft.Performance.Toolkit.Plugins.Core.Packaging.Metadata;
+
+namespace Microsoft.Performance.Toolkit.Plugins.Core.Discovery
+{
+    /// <summary>
+    ///    A discoverer that is capable of discovering <see cref="AvailablePluginInfo"/>.
+    /// </summary>
+    public interface IPluginDiscoverer
+    {
+        /// <summary>
+        ///     Discovers the latest version of all plugins.
+        /// </summary>
+        /// <param name="cancellationToken">
+        ///     Signals that the caller wishes to cancel the operation.
+        /// </param>
+        /// <returns>
+        ///     The <see cref="AvailablePluginInfo"/> of all discovered plugin grouped by plugin identifiers.
+        /// </returns>
+        /// TODO: 243 Add search
+        Task<IReadOnlyDictionary<string, AvailablePluginInfo>> DiscoverPluginsLatestAsync(CancellationToken cancellationToken);
```

Comment:
Why is this a dictionary? What are the keys supposed to be again?

Reply:
The keys are `PluginIdentity.Id`s. This is to make sure each plugin id gets only "one latest" `AvailablePluginInfo`. I'll update the docstring so that it's not confusing.

Reply:
I feel like we should just ditch the dictionary altogether since I'm not sure what it adds
1. Clients will likely just completely ignore the keys
2. The method spec doesn't actually enforce anything. Implementations can return a dictionary with bogus values for keys, and different versions of the same plugin as values. Whoever actually consumes this method should be responsible for filtering out duplicate results

Reply:
Case in point: `IPluginsDiscoveryOrchestrator` doesn't return a dictionary

IMO, this return signature just makes correct implementations awkward. Most likely an implementation will just be doing something like
```cs
return latestVersions.ToDictionary(p => p.Identity.Id, p => p);
```

Reply:
Updated